packages/web/src/pages/saved-page/components/desktop/AlbumsTabPage.tsx

```diff
@@ -37,10 +37,12 @@ export const AlbumsTabPage = () => {
     collections: albums
   } = useCollectionsData({ collectionType: 'album' })
 
-  const emptyAlbumsHeader = useSelector((state: CommonState) => {
-    const selectedCategory = getCategory(state, {
+  const selectedCategory = useSelector((state: CommonState) => {
```

Comment:
actually... although this is what we agreed to, I think this is probably wrong.
instead of only showing them in the purchase tab/category, we likely want to show purchased yet hidden tracks/albums on all tabs where they are relevant e.g. 'all'/'saved'/'purchased' if it was a purchased track that user saved (or that was autosaved).

seems like we should only filter out `if is_unlisted and user did not previously purchase`.

sthg like `... .filter(track => !track.is_unlisted || (isUsdcGated(streamConditions) and userHasAccess))`

same logic below for tracks tab in library

Reply:
added changes in DN to handle this for tracks, albums, and playlists